src/Silverback.Integration.Kafka.Testing/Messaging/Broker/Kafka/Mocks/InMemoryPartition.cs

```diff
@@ -54,6 +56,20 @@ public bool TryPull(Offset offset, out ConsumeResult<byte[]?, byte[]?>? result)
             {
                 if (offset.Value > LastOffset || _messages.Count == 0)
                 {
+                    if (_lastReportedOffsetForEndOfPartition < offset.Value && _messages.Count > _lastReportedOffsetForEndOfPartition ||
+                        _lastReportedOffsetForEndOfPartition == null && _messages.Count == 0)
+                    {
+                        result = new ConsumeResult<byte[]?, byte[]?>
```

Comment:
The real Kafka only sends the EOF message when explicitly enabled in the configuration. This mock should do the same.